NYPLAudiobookToolkit/DRM/FeedbookDRMProcessor.swift

```diff
@@ -117,28 +117,25 @@ class FeedbookDRMProcessor {
             var digestBytes = [UInt8](repeating: 0, count: Int(CC_SHA256_DIGEST_LENGTH))
             RSAUtils.SHA256HashedData(from: (licenseData as NSData)).getBytes(&digestBytes, length: Int(CC_SHA256_DIGEST_LENGTH))
             
-            var signatureBytes = [UInt8](repeating: 0, count: blockSize)
-            var signatureDataLength = blockSize
-            
-            let status = SecKeyRawSign(privateSecKey,
-                                       .PKCS1SHA256,
-                                       digestBytes,
-                                       digestBytes.count,
-                                       &signatureBytes,
-                                       &signatureDataLength)
-            
-            guard status == noErr else {
-                ATLog(.error, "Failed to sign data - \(status.description)")
+            guard let signatureData = Data(base64Encoded: signatureValue) else {
+                ATLog(.error, "Error decoding signature value")
                 return false
             }
+            var signatureBytes = Array(signatureData)
+            let signatureDataLength = signatureBytes.count
             
-            let signatureData = Data(bytes: signatureBytes, count: signatureBytes.count)
+            let status = SecKeyRawVerify(publicSecKey, .PKCS1SHA256, digestBytes, digestBytes.count, &signatureBytes, signatureDataLength)
             
-            guard signatureData.base64EncodedString() == signatureValue else {
-                ATLog(.error, "Signature does not match, DRM check failed")
+            guard status == noErr else {
+                if #available(iOS 11.3, *) {
+                    var errorMessage = ""
+                    SecCopyErrorMessageString(status, &errorMessage)
+                    ATLog(.error, "Failed to sign data - \(errorMessage)")
```

Comment:
this should be "verify" not "sign", right?

Also can you add some context to this log, such as do what this is sign/verification for, I assume Feedbooks? Otherwise when we read it in logs it would be unclear.

Same thing for the next log at line 135.

Reply:
That's right, should be public instead of private and verify instead of sign, fixed that

Reply:
Log output includes filename and line, so should be visible as "[ERROR] FeedbookDRMProcessor.swift:123: something wrong happened"